kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -46,14 +46,19 @@ impl<'tcx> GotocCtx<'tcx> {
             let base_name = self.codegen_var_base_name(&lc);
             let name = self.codegen_var_name(&lc);
             let ldata = &ldecls[lc];
-            let t = self.monomorphize(ldata.ty);
-            let t = self.codegen_ty(t);
+            let var_ty = self.monomorphize(ldata.ty);
+            let var_type = self.codegen_ty(var_ty);
             let loc = self.codegen_span(&ldata.source_info.span);
             // Indices [1, N] represent the function parameters where N is the number of parameters.
-            let sym =
-                Symbol::variable(name, base_name, t, self.codegen_span(&ldata.source_info.span))
-                    .with_is_hidden(!ldata.is_user_variable())
-                    .with_is_parameter(idx > 0 && idx <= num_args);
+            // Except that ZST fields are not included as parameters.
+            let sym = Symbol::variable(
+                name,
+                base_name,
+                var_type,
+                self.codegen_span(&ldata.source_info.span),
+            )
+            .with_is_hidden(!ldata.is_user_variable())
+            .with_is_parameter((idx > 0 && idx <= num_args) && !self.is_zst(var_ty));
```

Comment:
Does this revert when we switch to `fn_abi...`?

Reply:
This check will change, but we will still need to set to `false` things that are ignored parameters.